src/main.ts

```diff
@@ -81,6 +81,19 @@ export function buildObject(adapterInstance: ioBroker.Adapter, options: BuildObj
 		definition.common.desc = options.description;
 	}
 
+	if (definition.type === "state"){
```

Comment:
I would leave a comment above this block explaning what this is supposed to do:
The templates may contain arrays for some values that are no arrays in ioBroker. We find them and pin them to a single value.